### ISSUE TITLE
`Declaration expected` when using decorators

### ISSUE THREAD

**Issue by warpdesign** [microsoft/TypeScript#35910]
- VSCode Version: 1.41.1
- OS Version: OSX 10.15.2

Steps to Reproduce:

1. add `experimentalDecorators: true` to tsconfig.json
2. Use a decorator in a .js file

Here is the contents of my jsconfig.json:

```json
{
    "compilerOptions": {
      "allowSyntheticDefaultImports": false,
      "experimentalDecorators": true,
      "baseUrl": ".",
      "paths": {
        "$src/*": ["./src/*"],
        "$config/*": ["./config/*"]
      }
    },
    "exclude": ["node_modules", "dist"]
  }
```

![image](https://user-images.githubusercontent.com/199648/71483909-33ad4980-280a-11ea-9f7e-0dfa6ffee6b2.png)

Example code with decorator:

```js
import { authRequired } from './$src/authRequired';

const service = {
   @authRequired
   getPrice() {
       //...
   }
}
```

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/5770

**Comment by IkeyBenz** [comment]
That is a reference to an issue from *five* years ago... this hasn't been resolved yet? All I got from #5770 is that decorators don't work yet in Typescript, while Typescript's documentation says otherwise. Some clarity as to why @warpdesign 's example code doesn't work would be greatly appreciated.

**Comment by EwanValentine** [comment]
Yep ^ this would be useful clarification, I assumed because of projects like Angular it was something that worked out of the box already, but following examples from the TS docs, it doesn't seem to work as expected
